source/agora/node/FullNode.d

```diff
@@ -153,30 +153,13 @@ public class FullNode : API
     ***************************************************************************/
 
     public void start ()
-    {
-        this.startPeriodicDiscovery();
-        this.network.startPeriodicCatchup(this.ledger);
-    }
-
-    /***************************************************************************
-
-        Starts the periodic network discovery task.
-
-    ***************************************************************************/
-
-    private void startPeriodicDiscovery ()
     {
         import core.time;
 
-        this.taskman.runTask(
-        ()
-        {
-            while (1)
-            {
-                this.network.discover();
-                this.taskman.wait(5.seconds);
-            }
-        });
+        scope void discover () { this.network.discover(); }
```

Comment:
This is wrong. It should not be a `scope` delegate. If it's scope, it will be allocated on the stack. So when `start()` returns and the timer tries to call `discover`, it may crash.

Reply:
In fact, I was worried.